### ISSUE TITLE
Unable to connect to BIRDv4 socket: dial unix /var/run/calico/bird.ctl: connect: connection refused

### ISSUE THREAD

**Issue by GsssC** [projectcalico/calico#4603]
## Expected Behavior

Calico run well
## Current Behavior

Calico-node pods are unready. One of them continuesly log event `Unable to connect to BIRDv4 socket: dial unix /var/run/calico/bird.ctl: connect: connection refused` like below:
![image](https://user-images.githubusercontent.com/29768082/117811295-f3c9f380-b292-11eb-9785-4801b481f181.png)

## Possible Solution

Sorry I don't know
## Your Environment

* Calico version: v3.19.0
* Orchestrator version (e.g. kubernetes, mesos, rkt): 1.19.4
* Operating System and version: Linux n82 3.10.0-1160.2.2.el7.x86_64 #1 SMP Tue Oct 20 16:53:08 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**Comment by hextrim** [comment]
I have exactly the same problem. This is a wired issue.
I added node to a k8s cluster via kubespray.
It was working fine.
I shutdown cluster for the night and started in the morning the node I added to the cluster via kubespray has exactly the same problem as explained above.

Further:
[root@ht-prov-01 ~]# k logs -f pods/calico-node-2f497 -n kube-system
Error from server: Get "https://192.168.1.67:10250/containerLogs/kube-system/calico-node-2f497/calico-node?follow=true": dial tcp 192.168.1.67:10250: connect: no route to host

Basically calico is not operational on the node anymore.

Any idea how could this happen?

**Comment by hextrim** [comment]
Ignore me - schoolboy error - new node has also firewalld enabled calico ports and k8s components port comms.

**Comment by mbrancato** [comment]
To tag onto this, I had a similar error. The cause was that during a certificate rotation, the Secret `calico-etcd-secrets` was not updated with the latest etcd certs.

**Comment by masrur-qr** [comment]
sudo firewall-cmd --permanent --add-port=179/tcp

sudo firewall-cmd --permanent --add-port=4789/udp

sudo firewall-cmd --permanent --add-port=6443/tcp

sudo firewall-cmd --reload

don't forget to install "firewall"

sudo systemctl stop firewalld
sudo systemctl disable firewalld


it solved my problem with BIRD connection refused
